File: backend/src/main/java/moe/mewore/rabbit/backend/simulation/data/FrameDataType.java
package moe.mewore.rabbit.backend.simulation.data;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum FrameDataType {

    BYTE(Byte.BYTES),
    SHORT(Short.BYTES),
    INTEGER(Integer.BYTES),
    LONG(Long.BYTES),
    FLOAT(Float.BYTES),
    DOUBLE(Double.BYTES),
    VECTOR3F(3 * Float.BYTES);

    @Getter(AccessLevel.PACKAGE)
    private final int byteSize;
}

File: frontend/src/game/simulation/data/frame-data-type.ts
export class FrameDataType {
    constructor(readonly byteSize: number) {}
}

export const BYTE = new FrameDataType(1);
export const INTEGER = new FrameDataType(4);
export const LONG = new FrameDataType(8);
export const FLOAT = new FrameDataType(4);
export const DOUBLE = new FrameDataType(8);
export const VECTOR3F = new FrameDataType(3 * FLOAT.byteSize);
